lib/Elastica/Aggregation/Filter.php

```diff
@@ -13,13 +14,19 @@ class Filter extends AbstractAggregation
 {
     /**
      * @param string         $name
-     * @param AbstractFilter $filter
+     * @param AbstractQuery $filter
      */
-    public function __construct($name, AbstractFilter $filter = null)
+    public function __construct($name, $filter = null)
     {
         parent::__construct($name);
 
         if ($filter !== null) {
+            if ($filter instanceof AbstractFilter) {
```

Comment:
The other advantage is that we don't have to check for null every time. The deprecated message will be shown anyways if someone uses a filter, as somehow the class has to be initiated.